src/query/functions/temporal/aggregation_test.go

```diff
@@ -0,0 +1,207 @@
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/executor/transform"
+	"github.com/m3db/m3/src/query/parser"
+	"github.com/m3db/m3/src/query/test"
+	"github.com/m3db/m3/src/query/test/executor"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	description    string
+	opType         string
+	afterBlockOne  [][]float64
+	afterAllBlocks [][]float64
+}
+
+var testCases = []testCase{
+	{
+		description: "testing avg_over_time",
+		opType:      AvgTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 2.5},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 7},
+		},
+		afterAllBlocks: [][]float64{
+			{2, 2, 2, 2, 2},
+			{7, 7, 7, 7, 7},
+		},
+	},
+	{
+		description: "testing count_over_time",
+		opType:      CountTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 4},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 5},
+		},
+		afterAllBlocks: [][]float64{
+			{5, 5, 5, 5, 5},
+			{5, 5, 5, 5, 5},
+		},
+	},
+	{
+		description: "testing min_over_time",
+		opType:      MinTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 1},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 5},
+		},
+		afterAllBlocks: [][]float64{
+			{0, 0, 0, 0, 0},
+			{5, 5, 5, 5, 5},
+		},
+	},
+	{
+		description: "testing max_over_time",
+		opType:      MaxTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 4},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 9},
+		},
+		afterAllBlocks: [][]float64{
+			{4, 4, 4, 4, 4},
+			{9, 9, 9, 9, 9},
+		},
+	},
+	{
+		description: "testing sum_over_time",
+		opType:      SumTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 10},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 35},
+		},
+		afterAllBlocks: [][]float64{
+			{10, 10, 10, 10, 10},
+			{35, 35, 35, 35, 35},
+		},
+	},
+	{
+		description: "testing stddev_over_time",
+		opType:      StdDevTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 1.1180},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 1.4142},
+		},
+		afterAllBlocks: [][]float64{
+			{1.4142, 1.4142, 1.4142, 1.4142, 1.4142},
+			{1.4142, 1.4142, 1.4142, 1.4142, 1.4142},
+		},
+	},
+	{
+		description: "testing stdvar_over_time",
+		opType:      StdVarTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 1.25},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 2},
+		},
+		afterAllBlocks: [][]float64{
+			{2, 2, 2, 2, 2},
+			{2, 2, 2, 2, 2},
+		},
+	},
+}
+
+// B1 has NaN in first series, first position
+func TestAggregation(t *testing.T) {
+	for _, testCase := range testCases {
+		values, bounds := test.GenerateValuesAndBounds(nil, nil)
```

Comment:
Unfortunately if you're using this as your `values`, you only get a really nice set of data points which do not help catch weird edge case issues

Reply:
Will add other test cases

Reply:
Yeah- I can do that in another pr